File: microservicio-cliente-persona/src/main/java/com/develop/clientepersona/entity/Persona.java
package com.develop.clientepersona.entity;

import com.develop.clientepersona.enums.GenderPerson;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.sql.Timestamp;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "personas")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE) // También puedes usar JOINED
@DiscriminatorColumn(name = "person_type", discriminatorType = DiscriminatorType.STRING)
public class Persona {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @CreationTimestamp
    @Column(name = "creation_date")
    private Timestamp creationDate;
    @Column(name = "full_name")
    private String fullName;
    @Enumerated(EnumType.STRING)
    @Column(name = "gender_person")
    private GenderPerson genderPerson;
    @Column(name = "age")
    private int age;
    @Column(name = "identification", unique = true)
    private String identification;
    @Column(name = "address")
    private String address;
    @Column(name = "phone")
    private String phone;
}

File: client-person/Models/Persona.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using client_person.Enum;

namespace client_person.Models
{
    public class Persona
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long id { get; set; }

        [Required]
        public string? fullName { get; set; }
        
        [Required]
        [EnumDataType(typeof(GenderPerson))]
        public GenderPerson genderPerson { get; set; }

        [Required]
        public int age { get; set; }

        [Required]
        public string? identification { get; set; }

        [Required]
        public string? address { get; set; }

        [Required]
        public string? phone { get; set; }

        public DateTime creationDate { get; set; }
    }
}